### PR TITLE
chore(deps): update build deps

### PR DESCRIPTION
**Pull request by Bad3r**

### CODE REVIEW

**Comment by Bad3r** [comment]
!build

**Comment by Bad3r** [comment]
@flathubbot build

**Comment by Bad3r** [comment]
bot, build com.logseq.Logseq

**Comment by Bad3r** [comment]
@kanru the build keeps failing because it can't resolve github.com any idea what is causing it?

**Comment by Bad3r** [comment]
bot, build com.logseq.Logseq

**Comment by Bad3r** [comment]
Hmm, It's trying to access the network for some reason during build (flatpak/flatpak#996). not sure what I changed that affected it.

**Comment by kanru** [comment]
> @kanru the build keeps failing because it can't resolve github.com any idea what is causing it?

Please check README.md or the CI script on how to update the manifest, or I can also check it later.

**Comment by Bad3r** [comment]
@kanru I tried to follow the steps in my local env and in a clean fedora env. The following command continues to fail:  
\`\`\`ps  
\[root@706f1603500c ls-flatpak\]# flatpak-node-generator -r yarn \\  
 logseq-$VERSION/yarn.lock --electron-node-headers -o generated-sources.json  
Found 28 lockfiles.  
Reading packages from lockfiles...  
Traceback (most recent call last):  
 File "/root/.local/bin/flatpak-node-generator", line 8, in \<module>  
   sys.exit(main())  
            ^^^^^^  
 File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak\_node\_generator/main.py", line 277, in main  
   asyncio.run(\_async\_main())  
 File "/usr/lib64/python3.11/asyncio/runners.py", line 190, in run  
   return runner.run(main)  
          ^^^^^^^^^^^^^^^^  
 File "/usr/lib64/python3.11/asyncio/runners.py", line 118, in run  
   return self.\_loop.run\_until\_complete(task)  
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  
 File "/usr/lib64/python3.11/asyncio/base\_events.py", line 653, in run\_until\_complete  
   return future.result()  
          ^^^^^^^^^^^^^^^  
 File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak\_node\_generator/main.py", line 198, in \_async\_main  
   packages.update(lockfile\_provider.process\_lockfile(lockfile))  
 File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak\_node\_generator/providers/yarn.py", line 107, in process\_lockfile  
   yield self.process\_package(lockfile, name\_line, package)  
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  
 File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak\_node\_generator/providers/yarn.py", line 92, in process\_package  
   if self.is\_git\_version(entry\['resolved'\]):  
                          ~~~~~^^^^^^^^^^^^  
\`\`\`  
  
Any idea what is the cause of this issue?  
I can't find any missing resolve statistics in yarn.lock

File: com.logseq.Logseq.yml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+runtime-version: "22.08"
 sdk: org.freedesktop.Sdk
 base: org.electronjs.Electron2.BaseApp
-base-version: '22.08'
+base-version: "22.08"
 sdk-extensions:
-  - org.freedesktop.Sdk.Extension.node16
-  - org.freedesktop.Sdk.Extension.openjdk8
+  - org.freedesktop.Sdk.Extension.node18
+  - org.freedesktop.Sdk.Extension.openjdk17
```
Comment (kanru) [inline com.logseq.Logseq.yml:10]: In `logseq/.github/workflows/build.yml` JAVA_VERSION is 11

File: com.logseq.Logseq.yml [hunk 2]
```diff
@@ -23,39 +24,39 @@ modules:
...
-        ENABLE_FILE_SYNC_PRODUCTION: 'true'
+        npm_config_offline: "true"
+        ENABLE_PLUGINS: "true"
+        ENABLE_FILE_SYNC_PRODUCTION: "true"
     sources:
       - type: file
-        url: https://download.clojure.org/install/linux-install-1.10.1.763.sh
-        sha256: 91421551872d421915c4a598741aefcc6749d3f4aafca9c08f271958e5456e2c
+        url: https://download.clojure.org/install/linux-install-1.11.1.1273.sh
+        sha256: 4938afe6c0690d2a6553141857742d94a7350c02b4fa57cd6a9c1b7cbe66492e
```
Comment (kanru) [inline com.logseq.Logseq.yml:41]: In `logseq/.github/workflows/build.yml` CLOJURE_VERSION is still 1.10.1.763

File: com.logseq.Logseq.yml [hunk 3]
```diff
@@ -23,39 +24,39 @@ modules:
...
       - type: file
-        url: https://download.clojure.org/install/clojure-tools-1.10.1.763.tar.gz
-        sha256: 2a3ec8a6c3639035c2bba10945ae9007ab0dc9136766b95d2161f354e62a4d10
+        url: https://download.clojure.org/install/clojure-tools-1.11.1.1273.tar.gz
+        sha256: 5f8bafcf24bd148ac9bcbe60a8e7b8093c9ed8eb8e0f387199770e38f0e40ce6
       - type: git
         url: https://github.com/borkdude/rewrite-edn
-        commit: edd87dc7f045f28d7afcbfc44bc0f0a2683dde62
+        commit: f9e71baca3122e04dfc83e14f0ef63da219b0ee6 # v0.4.6
         dest: libs/rewrite-edn
```
Comment (kanru) [inline com.logseq.Logseq.yml:48]: If in `deps.edn` it's no longer pinned to git version then we can remove this. `maven-sources.json` will take care of it.

File: com.logseq.Logseq.yml [hunk 4]
```diff
@@ -23,39 +24,39 @@ modules:
...
       - type: git
-        url: https://github.com/lread/rewrite-cljc-playground
-        commit: ba1e4255cd1c1fd09dbe2e90556bfa66d5459234
-        dest: libs/rewrite-cljc
+        url: https://github.com/clj-commons/rewrite-clj
+        commit: d0c611e2da6aaddb46ae4f1f0c9531e4b4d3ce7a # v1.1.47
+        dest: libs/rewrite-clj
```
Comment (kanru) [inline com.logseq.Logseq.yml:52]: Same for this one, it looks like `rewride-edn` does  not use git dependency so we can remove this.

File: com.logseq.Logseq.yml [hunk 5]
```diff
@@ -23,39 +24,39 @@ modules:
...
       - type: git
         url: https://github.com/logseq/cljs-time
         commit: 5704fbf48d3478eedcf24d458c8964b3c2fd59a9
         dest: libs/cljs-time
       - type: git
         url: https://github.com/clj-commons/hickory
-        commit: a308fafdf1e0483087a105544f2cb190cc3289b0
+        commit: bef62d990d070868bb2181413b80c4d2e1c23fbf # v0.7.3
         dest: libs/hickory
```
Comment (kanru) [inline com.logseq.Logseq.yml:60]: We can also remove this

File: com.logseq.Logseq.yml [hunk 6]
```diff
@@ -82,16 +83,16 @@ modules:
...
       # disable clojure tools online fetching
-      - unzip /usr/local/lib/clojure/libexec/clojure-tools-1.10.1.763.jar clojure/tools/gitlibs/impl.clj
+      - unzip /usr/local/lib/clojure/libexec/clojure-tools-1.11.1.1273.jar clojure/tools/gitlibs/impl.clj
       - sed -i 's/git-fetch git-dir/do/' clojure/tools/gitlibs/impl.clj
-      - zip -u /usr/local/lib/clojure/libexec/clojure-tools-1.10.1.763.jar clojure/tools/gitlibs/impl.clj
+      - zip -u /usr/local/lib/clojure/libexec/clojure-tools-1.11.1.1273.jar clojure/tools/gitlibs/impl.clj
       # setup offline cache
       - git clone --bare libs/rewrite-edn .gitlibs/_repos/github.com/borkdude/rewrite-edn
-      - git clone --bare libs/rewrite-cljc .gitlibs/_repos/github.com/lread/rewrite-cljc-playground
+      - git clone --bare libs/rewrite-clj .gitlibs/_repos/github.com/clj-commons/rewrite-clj
```
Comment (kanru) [inline com.logseq.Logseq.yml:95]: can be removed

File: com.logseq.Logseq.yml (hunk 1 above)
Comment (Bad3r) [reply com.logseq.Logseq.yml:10]: @kanru it is. They have been ignoring updating deps until they automate the process. I have tested and used logseq with the LTS versions for Node and openjdk for more than 3 months. There is no issue in upgrading, and staying on outdated versions is not a good idea especially when there is no need to.

File: com.logseq.Logseq.yml (hunk 3 above)
Comment (Bad3r) [reply com.logseq.Logseq.yml:48]: @kanru isn't it better for security to have it pinned?

Comment (Bad3r) [reply com.logseq.Logseq.yml:48]: @kanru do you mean remove the commit hash or the dep completely?

Comment (kanru) [reply com.logseq.Logseq.yml:48]: The entire dep can be removed, and later we don't need to link them to the cache path.

**Comment by Bad3r** [comment]
> @kanru I tried to follow the steps in my local env and in a clean fedora env. The following command continues to fail:
> 
> ```python
> [root@706f1603500c ls-flatpak]# flatpak-node-generator -r yarn \
>   logseq-$VERSION/yarn.lock --electron-node-headers -o generated-sources.json
> Found 28 lockfiles.
> Reading packages from lockfiles...
> Traceback (most recent call last):
>   File "/root/.local/bin/flatpak-node-generator", line 8, in <module>
>     sys.exit(main())
>              ^^^^^^
>   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/main.py", line 277, in main
>     asyncio.run(_async_main())
>   File "/usr/lib64/python3.11/asyncio/runners.py", line 190, in run
>     return runner.run(main)
>            ^^^^^^^^^^^^^^^^
>   File "/usr/lib64/python3.11/asyncio/runners.py", line 118, in run
>     return self._loop.run_until_complete(task)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib64/python3.11/asyncio/base_events.py", line 653, in run_until_complete
>     return future.result()
>            ^^^^^^^^^^^^^^^
>   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/main.py", line 198, in _async_main
>     packages.update(lockfile_provider.process_lockfile(lockfile))
>   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/providers/yarn.py", line 107, in process_lockfile
>     yield self.process_package(lockfile, name_line, package)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/providers/yarn.py", line 92, in process_package
>     if self.is_git_version(entry['resolved']):
>                            ~~~~~^^^^^^^^^^^^
> KeyError: 'resolved'
> ```
> 
> Any idea what is the cause of this issue? I can't find any missing resolve statistics in yarn.lock

@kanru I made the changes suggested, and the build now fails for another vague reason. I am still unable to build it locally due to the error above.

File: com.logseq.Logseq.yml [hunk 7]
```diff
@@ -83,16 +72,14 @@ modules:
     build-commands:
       # setup clojure
       - sed -i 's/curl/#curl/' linux-install-1.10.1.763.sh
-      - chmod +x linux-install-1.10.1.763.sh
+      - chmod 755 linux-install-1.10.1.763.sh
       - ./linux-install-1.10.1.763.sh
       # disable clojure tools online fetching
-      - unzip /usr/local/lib/clojure/libexec/clojure-tools-1.10.1.763.jar clojure/tools/gitlibs/impl.clj
+      - unzip /usr/local/lib/clojure/libexec/linux-install-1.10.1.763.jar clojure/tools/gitlibs/impl.clj
```
Comment (kanru) [inline com.logseq.Logseq.yml:78]: Why do you have to change this and the next chunk?

**Comment by kanru** [comment]
> > @kanru I tried to follow the steps in my local env and in a clean fedora env. The following command continues to fail:
> > 
> > ```python
> > [root@706f1603500c ls-flatpak]# flatpak-node-generator -r yarn \
> >   logseq-$VERSION/yarn.lock --electron-node-headers -o generated-sources.json
> > Found 28 lockfiles.
> > Reading packages from lockfiles...
> > Traceback (most recent call last):
> >   File "/root/.local/bin/flatpak-node-generator", line 8, in <module>
> >     sys.exit(main())
> >              ^^^^^^
> >   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/main.py", line 277, in main
> >     asyncio.run(_async_main())
> >   File "/usr/lib64/python3.11/asyncio/runners.py", line 190, in run
> >     return runner.run(main)
> >            ^^^^^^^^^^^^^^^^
> >   File "/usr/lib64/python3.11/asyncio/runners.py", line 118, in run
> >     return self._loop.run_until_complete(task)
> >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/usr/lib64/python3.11/asyncio/base_events.py", line 653, in run_until_complete
> >     return future.result()
> >            ^^^^^^^^^^^^^^^
> >   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/main.py", line 198, in _async_main
> >     packages.update(lockfile_provider.process_lockfile(lockfile))
> >   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/providers/yarn.py", line 107, in process_lockfile
> >     yield self.process_package(lockfile, name_line, package)
> >           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/root/.local/pipx/venvs/flatpak-node-generator/lib64/python3.11/site-packages/flatpak_node_generator/providers/yarn.py", line 92, in process_package
> >     if self.is_git_version(entry['resolved']):
> >                            ~~~~~^^^^^^^^^^^^
> > KeyError: 'resolved'
> > ```
> > 
> > Any idea what is the cause of this issue? I can't find any missing resolve statistics in yarn.lock
> 
> @kanru I made the changes suggested, and the build now fails for another vague reason. I am still unable to build it locally due to the error above.

Logseq version didn't change so we don't need to update the node dependencies. You can skip this step.

The error might be due to incorrectly generated lock file or other reason.

**Comment by Bad3r** [comment]
bot, build com.logseq.Logseq

**Comment by Bad3r** [comment]
bot, build com.logseq.Logseq

**Comment by Bad3r** [comment]
@kanru I reverted back the node version. Running flatpak-node-generator still fails for me locally. The build fails because it keeps trying to download node

**Comment by Bad3r** [comment]
@kanru there must be something missing from the instructions. I have tested in clean VMs to follow the steps and always the same error shows up.

**Comment by kanru** [comment]
> @kanru there must be something missing from the instructions. I have tested in clean VMs to follow the steps and always the same error shows up.

I'll try to reproduce tonight.

**Comment by Bad3r** [comment]
@kanru this is the dockerfile I used for testing:
```dockerfile
# Dockerfile
FROM fedora:latest

RUN dnf update -y && \
    dnf upgrade -y && \
    dnf install -y npm python pip git && \
    dnf clean all

RUN npm install --global yarn && \
    pip install pipx

RUN git clone https://github.com/flatpak/flatpak-builder-tools && \
    cd flatpak-builder-tools/node/ && \
    pipx install .

VOLUME /ls-flatpak

CMD ["/bin/bash"]
```

This should take care of installing all pre-reqs

next to build and run the image:
```sh
$ docker build --tag fedora-flatpak:latest .
$ docker run --rm -it -v /path/to/repo:/logseq-flatpak fedora-flatpak:latest /bin/bash
$ cd /logseq-flatpak
```

Replace the path with your local repo path; this should be an easy way to confirm the issue.

**Comment by kanru** [comment]
Thanks for the docker file. We should also add flatpak dependencies

```
flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
flatpak install -y flathub org.freedesktop.Platform//22.08 org.freedesktop.Sdk//22.08
flatpak install -y flathub org.electronjs.Electron2.BaseApp/x86_64/22.08
```

Trying to build now..

**Comment by Bad3r** [comment]
@kanru dockerimage updated :D

**Comment by kanru** [comment]
> @kanru dockefile updated :D

missed two more extensions

```
flatpak install -y flathub org.freedesktop.Sdk.Extension.node16/x86_64/22.08
flatpak install -y flathub org.freedesktop.Sdk.Extension.openjdk11/x86_64/22.08
```

**Comment by Bad3r** [comment]
@kanru aren't these defined in the manifest file? Why do they need to be installed separately?
```
runtime: org.freedesktop.Platform
runtime-version: "22.08"
sdk: org.freedesktop.Sdk
base: org.electronjs.Electron2.BaseApp
base-version: "22.08"
sdk-extensions:
  - org.freedesktop.Sdk.Extension.node16
  - org.freedesktop.Sdk.Extension.openjdk11

```

**Comment by Bad3r** [comment]
Adding these extensions keeps failing due to dbus not being accessible from within the docker image. Not sure if it's possible to build in docker without creating a local repo and caching it like what you already do to build the script in the CI.

**Comment by Bad3r** [comment]
but I think I can run these commands locally and I'll be able to build locally now on the host

**Pull request closed without merging**